### PR TITLE
For #207: Removing TargetScope and TargetGroup from Card

### LINKED ISSUE #207: Card.cs:18: Remove TargetScope and Group. Update...
The puzzle `1-fbf7b564` from EnigmaDragons/ProjectNeon#1 has to be resolved: 

https://github.com/EnigmaDragons/ProjectNeon/blob/b2cdc09c7b448a0b29cb5845ba1156a1aa32d223/src/ProjectNeon/Assets/Data/Templates/Card.cs#L18-L18

The puzzle was created by Silas Reinagel on 26-Sep-19. 

Estimate: 15 minutes,  role: DEV. 

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### PR DESCRIPTION
**Pull request by paulodamaso**
Closes #207: Removing TargetScope and TargetGroup from Card

- Removed TargetScope and TargetGroup from Card
- Fixed DumbAI to use Scope and Group from first CardAction instead of Card
- Added accessors to get these values from CardAction
- Changed PLayedCard to work with Target array

### CODE REVIEW

File: src/ProjectNeon/Assets/Data/Templates/CardAction.cs [hunk 1]
```diff
@@ -14,4 +14,15 @@ public void Apply(Target target)
             effect => effect.Apply(target)
         );
     }
+
+    public Scope Scope
```
Comment (SilasReinagel) [inline src/ProjectNeon/Assets/Data/Templates/CardAction.cs]: For these properties, use lambda syntax. It makes the code more terse.

File: src/ProjectNeon/Assets/Data/Templates/DumbAI.cs [hunk 2]
```diff
@@ -14,7 +14,7 @@ public override PlayedCard Play(Enemy activeEnemy)
     {
         var me = activeEnemy.AsMember();
         var card = activeEnemy.Deck.Cards.Random();
-        var possibleTargets = battleState.GetPossibleEnemyTeamTargets(me, card.TargetGroup, card.TargetScope);
+        var possibleTargets = battleState.GetPossibleEnemyTeamTargets(me, card.Actions[0].Group, card.Actions[0].Scope);
```
Comment (SilasReinagel) [inline src/ProjectNeon/Assets/Data/Templates/DumbAI.cs]: We need Targets for each Effect in Played card, not just the first.

File: src/ProjectNeon/Assets/Features/Battle/Cards/PlayedCard.cs [hunk 3]
```diff
@@ -1,12 +1,29 @@
...
-    public PlayedCard Init(Member performer, Target target, Card card)
+    /**
+     * Member which played the card
+     */
+    private Member performer;
+
+    /**
+     * Each Target targeted by the Card effects in the same order that the effects are stored in Card.
+     * 
+     * @todo #207:30min We need a better solution to link targets to effects.
```
Comment (SilasReinagel) [inline src/ProjectNeon/Assets/Features/Battle/Cards/PlayedCard.cs]: This puzzle isn't clear. Let's use clear and unambiguous language about what the exact problem is.

File: src/ProjectNeon/Assets/Features/Battle/Cards/SelectCardTargets.cs [hunk 4]
```diff
@@ -46,7 +46,7 @@ private void BeginSelection()
         cardPresenter.Set(_selectedCard, () => { });
         uiView.SetActive(true);
 
-        var possibleTargets = battleState.GetPossiblePlayerTargets(_selectedCard.TargetGroup, _selectedCard.TargetScope);
+        var possibleTargets = battleState.GetPossiblePlayerTargets(_selectedCard.Actions[0].Group, _selectedCard.Actions[0].Scope);
```
Comment (SilasReinagel) [inline src/ProjectNeon/Assets/Features/Battle/Cards/SelectCardTargets.cs]: Add a puzzle here for selected targets for actions after the first.

File: src/ProjectNeon/Assets/Data/Templates/CardAction.cs (hunk 1 above)
Comment (paulodamaso) [reply src/ProjectNeon/Assets/Data/Templates/CardAction.cs]: Done

File: src/ProjectNeon/Assets/Data/Templates/DumbAI.cs (hunk 2 above)
Comment (paulodamaso) [reply src/ProjectNeon/Assets/Data/Templates/DumbAI.cs]: @SilasReinagel My bad, I was thinking in a dumb ai just chosing the first target; I've fixed it

File: src/ProjectNeon/Assets/Features/Battle/Cards/PlayedCard.cs (hunk 3 above)
Comment (paulodamaso) [reply src/ProjectNeon/Assets/Features/Battle/Cards/PlayedCard.cs]: @SilasReinagel I think that it's better now, please take a look and say what yo think

File: src/ProjectNeon/Assets/Features/Battle/Cards/SelectCardTargets.cs (hunk 4 above)
Comment (paulodamaso) [reply src/ProjectNeon/Assets/Features/Battle/Cards/SelectCardTargets.cs]: @SilasReinagel Done!

**Pull request merged**